java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSubscriptionImpl.java

```diff
@@ -133,40 +142,132 @@ public void onCompleted() {
     }
 
     @Override
-    public BarrageTable entireTable() {
-        return partialTable(null, null, false);
+    public boolean isCompleted() {
+        return completed;
+    }
+
+    @Override
+    public synchronized void waitForCompletion() throws InterruptedException {
```

Comment:
Agreed, addressed by calling `notifyAll()` after creating `completedCondition` which should wake all threads and convert them to waiting on `completedCondition` instead of the instance lock.